### DISCUSSION TITLE
Links to headers not working when they contain special characters

### DISCUSSION THREAD

**Discussion by juancrrn** [requarks/wiki#5377] [Error / Bug Report] [answered]
### Scenario

I have created a page in Markdown with a header in Spanish:

```markdown
#### Este es un título
```

As you can see, the header contains a _special_ character `í` ([ASCII 161 lowercase letter i with acute accent](https://theasciicode.com.ar/extended-ascii-code/lowercase-letter-i-acute-accent-ascii-code-161.html)).

I am running version 2.5.170.

### What is working

Once rendered, when hovering the header with the mouse, a `¶` appears. If I click it, then the page is scrolled down to that secction.

### What is not working

If I right click the `¶`, copy the generated link and then go to a new tab and paste it, the scroll is not done. The generated link is: `https://{host}/{page}#este-es-un-título`.

### Extra analysis

The HTML tag generated for the header, which contains a generated id, is the following one:

```html
<h4 id="este-es-un-título" class="toc-header"><a href="#este-es-un-título" class="toc-anchor">¶</a> Este es un título</h4>
```

As you can see, the id contains the `í` symbol.

This happens with other _special_ characters as well.

### Temporal fix

If I remove the special characters from the title, it works just fine.

**Comment by NGPixel** [comment, marked as answer]
Update to latest version.

**Reply by juancrrn** [reply]
It was a complicated decision to upgrade, but that solved it. Thank you.
